src/main/java/core/dao/car/CarDaoJdbc.java

```diff
@@ -0,0 +1,215 @@
+package core.dao.car;
+
+import core.dao.DaoUtils;
+import core.lib.Dao;
+import core.model.Car;
+import core.model.DataProcessingException;
+import core.model.Driver;
+import core.model.ErrorMessages;
+import core.model.Manufacturer;
+import core.utils.ConnectionUtils;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarDaoJdbc implements CarDao {
+    private static final String INSERT_DRIVER_EXCEPTION =
+            "An error occurred while adding driver with id = %d";
+    private static final String REMOVE_DRIVER_EXCEPTION =
+            "An error occurred removing drivers for car with id = %d";
+    private static final String GET_DRIVERS_EXCEPTION =
+            "An error has occurred while retrieving data for car id %d";
+    private static final String GET_CARS_BY_DRIVER_EXCEPTION =
+            "An error has occurred while retrieving data for driver id %d";
+    
+    @Override
+    public Car add(Car car) {
+        String insertCar = "INSERT INTO cars(model, manufacturer)"
+                        + " VALUES(?, ?);";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement insertCarStatement = connection.prepareStatement(insertCar,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            insertCarStatement.setString(1, car.getModel());
+            insertCarStatement.setLong(2, car.getManufacturer().getId());
+            insertCarStatement.executeUpdate();
+            ResultSet resultSet = insertCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject("id", Long.class));
+            }
+            insertDrivers(car, connection);
```

Comment:
let's not pass the connection because we do not close the statement anyway
let's just move this line out of the `try` block and open a separate connection in `insertDrivers()` method

Reply:
changed it in the way that two statements aren't opened at once